packages/next/src/server-build.ts

```diff
@@ -2085,6 +2085,18 @@ export async function serverBuild({
           ]
         : []),
 
+      // before processing rewrites, remove any special `/index` routes that were added
+      // as these won't be properly normalized by `afterFilesRewrites`
+      ...(afterFilesRewrites.length > 0
+        ? [
+            {
+              src: path.posix.join('/', entryDirectory, '/index(.action|.rsc)'),
```

Comment:
```suggestion
              src: path.posix.join('/', entryDirectory, '/index(\\.action|\\.rsc)'),
```
should we escape these so they aren't treated as greedy matches